features/cellular/framework/device/CellularDevice.cpp

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2018, Arm Limited and affiliates.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "CellularDevice.h"
+#include "EventQueue.h"
+#include "CellularUtil.h"
+
+#ifdef CELLULAR_DEVICE
+#include CELLULAR_STRINGIFY(CELLULAR_DEVICE.h)
+#endif // CELLULAR_DEVICE
+
+namespace mbed {
+
+#ifdef CELLULAR_DEVICE
+CellularDevice *CellularDevice::get_default_instance()
```

Comment:
This should be marked MBED_WEAK so that someone can override it. And there should also be a (weak) definition that returns NULL if CELLULAR_DEVICE is not defined.

Both of those would fit the common pattern of default instances, even if maybe you wouldn't be intending to take advantage.

Reply:
Fixed.